packages/yew/src/callback.rs

```diff
@@ -82,4 +82,70 @@ impl<IN: 'static, OUT: 'static> Callback<IN, OUT> {
     }
 }
 
+impl<IN: 'static> Callback<IN> {
+    /// Creates a new callback from another callback and a function
+    /// That when emitted will call that function and will emit the original callback if it is
+    /// not `None`.
+    pub fn filter_reform<F, T>(&self, func: F) -> Callback<T>
```

Comment:
One thing I almost forgot: Would it make sense to have this be `Callback<T, Option<OUT>>` instead?

Reply:
Hm, that sounds like an interesting idea. I will try to amend the PR with this.

Reply:
Looks good. I amended the PR.